json_serialization_writer.go

```diff
@@ -614,14 +607,29 @@ func (w *JsonSerializationWriter) WriteCollectionOfInt8Values(key string, collec
 
 // GetSerializedContent returns the resulting byte array from the serialization writer.
 func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
-	w.trimLastPropertySeparator()
-	resultStr := strings.Join(w.writer, "")
-	return []byte(resultStr), nil
+	written := w.writer.Bytes()
+	writtenLen := len(written)
+	trimmed := make([]byte, 0, writtenLen)
+
+	sliceStart := 0
+	for _, idx := range w.separatorIndices {
+		trimmed = append(trimmed, written[sliceStart:idx]...)
```

Comment:
if you sorted the array by desc instead, wouldn't you need to perform fewer slicing and not need to track the slice start anymore?

Reply:
I don't think I'm fully following you. I'm guessing you're getting at something simple, but everything I'm imagining seems complicated, e.g., prepending or reversing `trimmed` or requiring you to know how long the ultimate string will be. Could you provide some pseudo code, maybe?

Reply:
sure.
1. sort the indices (descending order) before the loop
2. assign the buffer to trimmed
3. loop on the sorted indices
4. check the value in the trimmed at idx +1
5. if the value is } or ] reassign trimmed to `append(trimmed[0:idx-1], trimmed[idx+1:])`
6. done!